### PR TITLE
DOC: Added note about groupby excluding Decimal columns by default

### LINKED ISSUE #13821: 'groupby' multiple columns and 'sum' multiple columns with different types
#### Code Sample, a copy-pastable example if possible

```
from decimal import *
import pandas as pd
df = pd.DataFrame({'name': ['foo', 'bar', 'foo', 'bar'], 'title': ['boo', 'far', 'boo', 'far'], 'id': [123, 456, 123, 456], 'int_column': [1, 2, 3, 4], 'dec_column1': [Decimal(0.50), Decimal(0.15), Decimal(0.25), Decimal(0.40)], 'dec_column2': [Decimal(0.20), Decimal(0.30), Decimal(0.55), Decimal(0.60)]})
df.groupby(['name', 'title', 'id'], as_index=False).sum()
```

#### Expected Output

i have dataframe that looks something like this...

| **name** | **title** | **id** | **int_column** | **dec_column1** | **dec_column2** |

...that has multiple rows with the same **name**, **title**, and **id**, but different values for the 3 number columns (**int_column**, **dec_column1**, **dec_column2**).
**int_column** == column of integers
**dec_column1** == column of decimals
**dec_column2** == column of decimals
I would like to be able to groupby the first three columns, and sum the last 3.  I would expect to be able to do the following:

df = df.groupby(['name', 'title', 'id'], as_index=False).sum()

however, the only column that gets summed and ends up in the final dataframe is the int_column.

| **name** | **title** | **id** | **int_column** |

if i explicitly name the columns, i can get the statement to target the decimal columns either on their own or together....

df = df.groupby(['name', 'title', 'id'], as_index=False)['dec_column1'].sum()
returns...
| **name** | **title** | **id** | **dec_column1** |
and...
df = df.groupby(['name', 'title', 'id'], as_index=False)['dec_column1', 'dec_column2'].sum()
returns...
| **name** | **title** | **id** | **dec_column1** | **dec_column1** |
however...
df = df.groupby(['name', 'title', 'id'], as_index=False)['dec_column1', 'dec_column2', 'user_num'].sum()
or...
df = df.groupby(['name', 'title', 'id'], as_index=False)['dec_column1', 'user_num', 'dec_column2'].sum()
or...
df = df.groupby(['name', 'title', 'id'], as_index=False)['user_num', 'dec_column1', 'dec_column2'].sum()
returns...
| **name** | **title** | **id** | **int_column** |

#### output of ``pd.show_versions()``

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.1.final.0
python-bits: 64
OS: Darwin
OS-release: 15.3.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.15.2
nose: 1.3.7
Cython: 0.22.1
numpy: 1.11.1
scipy: None
statsmodels: None
IPython: 5.0.0
sphinx: None
patsy: None
dateutil: 2.5.3
pytz: 2016.6.1
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.1
openpyxl: 2.3.5
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
rpy2: None
sqlalchemy: None
pymysql: 0.7.5.None
psycopg2: 2.5.5 (dt dec pq3 ext)

### PR DESCRIPTION
**Pull request by pdpark**
Also included example of how to explicitly aggregate by Decimal columns.

- closes #13821

### CODE REVIEW

File: doc/source/groupby.rst [hunk 1]
```diff
@@ -497,6 +497,28 @@ index are the group names and whose values are the sizes of each group.
 
    ``nth`` can act as a reducer *or* a filter, see :ref:`here <groupby.nth>`
 
+   Decimal columns are "nuisance" columns that .agg automatically excludes in groupby.
+
+   If you do wish to aggregate them you must do so explicitly:
```
Comment (gfyoung) [inline doc/source/groupby.rst]: Add a comma between "them" and "you"

File: doc/source/groupby.rst [hunk 2]
```diff
@@ -497,6 +497,28 @@ index are the group names and whose values are the sizes of each group.
 
    ``nth`` can act as a reducer *or* a filter, see :ref:`here <groupby.nth>`
 
+   Decimal columns are "nuisance" columns that .agg automatically excludes in groupby.
```
Comment (gfyoung) [inline doc/source/groupby.rst]: The word "nuisance" comes off a little too strong IMO.  Just state that they're excluded and (very briefly) why.

**Comment by topper-123** [comment]
I'm not so sure Pandas should even mention even ``Decimal`` on the ``groupby`` page.

Operations on ``Decimal``s will by nature be slow, and people could misunderstand the slowness as a slowness of Pandas. IMO Pandas should promote using the various numpy and pandas types and python ``str``, very seldomly other types.

Maybe it could go to the FAQ or maybe just to StackOverflow? If it goes in the pandas docs, there should IMO be a visible disclaimer regarding slowness and the backsides of doing math operations on python objects  rather than numpy objects.

File: doc/source/groupby.rst (hunk 2 above)
Comment (jreback) [reply doc/source/groupby.rst]: no, we call excluded columns ``nuisance`` so this is ok.

Comment (jreback) [inline doc/source/groupby.rst]: add this entire section in a ``.. note::`` block

File: doc/source/groupby.rst [hunk 3]
```diff
@@ -497,6 +497,28 @@ index are the group names and whose values are the sizes of each group.
...
+    dec = pd.DataFrame(
+            {'name': ['foo', 'bar', 'foo', 'bar'], 
+                'title': ['boo', 'far', 'boo', 'far'], 
+                'id': [123, 456, 123, 456], 
+                'int_column': [1, 2, 3, 4], 
+                'dec_column1': [Decimal('0.50'), Decimal('0.15'), Decimal('0.25'), Decimal('0.40')], 
+                'dec_column2': [Decimal('0.20'), Decimal('0.30'), Decimal('0.55'), Decimal('0.60')]
+            },
+        columns=['name','title','id','int_column','dec_column1','dec_column2']
+        )
```
Comment (jreback) [inline doc/source/groupby.rst]: show the dec here

File: doc/source/groupby.rst [hunk 4]
```diff
@@ -497,6 +497,28 @@ index are the group names and whose values are the sizes of each group.
...
+
+    dec.groupby(['name', 'title', 'id'], as_index=False).sum()
```
Comment (jreback) [inline doc/source/groupby.rst]: add a comment before each of the groupbys (e.g. ``.sum()`` by default excludes nuiscance), and how to include

File: doc/source/groupby.rst (hunk 2 above)
Comment (jreback) [inline doc/source/groupby.rst]: move this entire section down to 'Automatic exclusion of nuiscance columns'

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by pdpark** [comment]
Made requested changes.

**Comment by pdpark** [comment]
Fix for issue #17027.

**Comment by jreback** [comment]
can you rebase this

**Comment by jreback** [comment]
closing as stale

**Comment by gfyoung** [comment]
@jreback Can certainly rebase. 🙂

**Comment by jreback** [comment]
@gfyoung i am not sure this is in the right direction
needs significant edits

**Comment by jreback** [comment]
needs to be way way simpler

**Comment by jreback** [comment]
my point is i would start over

**Comment by pdpark** [comment]
I can revisit this.

**Review by jorisvandenbossche — COMMENTED** [review]
Some tips:

- I would leave the change to `gotchas.rst` as a separate PR, as this is really a different topic
- In the decimal example, try to trim down the code to what is minimally needed, to have a much simpler example. You don't need multiple decimal columns; a single key to group by is sufficient. 
- This is in general true for object dtype columns, not only for decimals. I would also mention that somewhere.

**Comment by alphaKuz** [comment]
Hello not interseted to learn

Sent from Outlook<http://aka.ms/weboutlook>
Thank you very much for replying
Jean K
________________________________

**Comment by pdpark** [comment]
@jorisvandenbossche Will do.

**Comment by pdpark** [comment]
@jorisvandenbossche & @jreback Requested changes made.

**Comment by jreback** [comment]
you need to merge master and force push, you are picking up all commits since july.

**Comment by jreback** [comment]
so I am not averse to expanding this section, however decimals DO now work on master if they are typed as EAs. So maybe a reference to that possibility here.

**Comment by datapythonista** [comment]
@pdpark do you have time to update based on the last comment?

**Comment by jorisvandenbossche** [comment]
> however decimals DO now work on master if they are typed as EAs

The decimals EA is not publicly exposed, only for testing purposes for now.

**Comment by jorisvandenbossche** [comment]
I updated this PR (slight rewording + simplified and fixed the example a little bit), should be ready to merge now.

**Comment by jorisvandenbossche** [comment]
> Some lines are >80, but I think in rst files we don't care

We care a little bit, to the extent that it fits in the code block box without generating scroll bars, but I think the limit is a bit higher than 80 for that.

**Comment by jorisvandenbossche** [comment]
Merging since it is a `/pandas/doc/source` change only (the CI is still an old version that is failing)

**Comment by jorisvandenbossche** [comment]
@pdpark Thanks a lot for the PR!

**Pull request merged**